[CL-1793] Flexible pages r2 e2e test fixes

lilkraftwerk opened the pull request:
All tests that failed on this run (https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19743/workflows/4e24c913-0a10-4be9-bddc-a9377b6b00c8/jobs/72906/steps) are working for me locally. Will still need to merge and test on the CI to be sure, but I think I've addressed all the issues

Fixes:

* Some selectors broke in other tests, I've fixed those
* The homepage hero banner tests were passing locally, but had probably failed the first time and I didn't notice. Basically, the form can only be submitted if an image is present, and you can't delete the image via a backend call, so the test can either be designed for the CI (starting from a clean slate, with no image) or for local re-runs (adapts to an existing image, since once it's added, it can't be removed). I opted to add an image as part of the payload in the `before` block, so the test always assumes one's there. The full functionality of the hero banner form is tested for custom pages, since a new page is created for each test it doesn't have the same issue.
* For the hero banner test, I moved the 3rd and 4th tests into the 2nd test block, since we want to check on the changes made in the test. This is more accurate, as tests should be independent, and having the 3rd test check for changes made in the 2nd could be flaky.

lilkraftwerk committed:
Drop image to appease hero banner validation

lilkraftwerk committed:
Fix homepage and custom page e2e tests

lilkraftwerk committed:
Fix edit profile e2e test

brentguf commented:
@lilkraftwerk 

> Will still need to merge and test on the CI to be sure, but I think I've addressed all the issues

You can run E2E tests on CI I think. I've tried it [here](https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19777/workflows/eb432ed4-c2ee-4812-bfd0-9b58a7bc8a4f). You can configure the citizenlab main repo branch to run this with, so I picked this one. It's described [here](https://www.notion.so/citizenlab/Testing-253d0c3cd99841a59929f7f615179935).

brentguf commented on front/cypress/e2e/admin/pages_and_menu/update_homepage_hero_banner_content.cy.ts:
```diff
@@ -164,26 +147,22 @@ describe('Admin: update Hero Banner content', () => {
 
-  it('views the signed-in content properly as a verified user', () => {
-    cy.setLoginCookie(verifiedEmail, verifiedPassword);
+    // log out, view page as verified user
```
Where do we log out and view the page as a verified user here? We just go to the landing page, right?

Btw, shall we rename goToLandingPage and e2e-landing-page to homepage equivalents now that we renamed the component?

lilkraftwerk replied:
Comments have been clarified

brentguf reviewed (commented):
> * Some selectors broke in other tests, I've fixed those
> * The homepage hero banner tests were passing locally, but had probably failed the first time and I didn't notice. Basically, the form can only be submitted if an image is present, and you can't delete the image via a backend call, so the test can either be designed for the CI (starting from a clean slate, with no image) or for local re-runs (adapts to an existing image, since once it's added, it can't be removed). I opted to add an image as part of the payload in the before block, so the test always assumes one's there. The full functionality of the hero banner form is tested for custom pages, since a new page is created for each test it doesn't have the same issue.

Looks good.

> For the hero banner test, I moved the 3rd and 4th tests into the 2nd test block, since we want to check on the changes made in the test. 

Sounds like you did the right thing. Some feedback: it would be easier if you made this comment in the code (Files changed tab) because I can't find the tests you're talking about. :-) Edit: found them!

lilkraftwerk committed:
Clarify comments in homepage test

lilkraftwerk commented:
> @lilkraftwerk
> 
> > Will still need to merge and test on the CI to be sure, but I think I've addressed all the issues
> 
> You can run E2E tests on CI I think. I've tried it [here](https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19777/workflows/eb432ed4-c2ee-4812-bfd0-9b58a7bc8a4f). You can configure the citizenlab main repo branch to run this with, so I picked this one. It's described [here](https://www.notion.so/citizenlab/Testing-253d0c3cd99841a59929f7f615179935).

Yup, I did that here: https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19768/workflows/14d6ea0c-787f-4dab-96f3-ab38ee75327f/jobs/73022/steps

Two tests related to our changes are still failing, I'll continue to investigate

lilkraftwerk committed:
Fix update homepage hero banner test

lilkraftwerk committed:
Custom page e2e test updates

lilkraftwerk commented:
New changes (will check on CI now): 

* Returned part of homepage hero banner test that closes the "complete your account" popup 
* Use another verified account (not the normal admin account) to check logged-in content
* Moved logic in before block of custom page test to main test block, deleted new page after all tests pass
* Made Cypress viewport slightly wider to make sure the whole menu bar is visible in tests

e: https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19786/workflows/c3790807-320f-4972-9781-0b13b5c3a0d7/jobs/73088

lilkraftwerk commented:
2/3 tests are fixed, going to simplify the homepage hero banner test now as it's the most problematic

lilkraftwerk committed:
Simplify homepage hero banner test

lilkraftwerk committed:
Wait for attachment to be uploaded before submitting

lilkraftwerk commented:
https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19826/workflows/51d9c31c-c5a6-4b19-b0e7-530d1f50c6e8/jobs/73272 tests are fixed

lilkraftwerk commented:
Two passes, I think should be all good. Will double check one last time after the final layout fixes but I don't think any of those should affect it https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab-ee/19856/workflows/7d40c87a-4849-4fed-8db3-d49e74ce2b6d/jobs/73413
